### ISSUE TITLE
ESQL: Allow PERCENTILE to accept a (constant) variable as second parameter

### ISSUE THREAD

**Issue by alex-spies** [elastic/elasticsearch#104965]
Analogous to https://github.com/elastic/elasticsearch/issues/104634 for `PERCENTILE`.

ESQL's `PERCENTILE` currently works well with foldable expressions, as long as they are in-line:
```
from employees | stats p50 = percentile(salary_change, -(50-1)+99);

->
p50:double
0.75
```
(C.f. [corresponding CSV test](https://github.com/elastic/elasticsearch/blob/50bafd306cbe7984c0f3694940dc2ef1aa4f0399/x-pack/plugin/esql/qa/testFixtures/src/main/resources/stats_percentile.csv-spec#L154))

This does not work if the expression comes from an eval:
```
from employees | stats p50 = percentile(salary_change, -(50-1)+99);

->
org.elasticsearch.xpack.esql.VerificationException: Found 1 problem
line 1:52: second argument of [percentile(salary_change, x)] must be a constant, received [x]
```

**Comment by costin** [comment]
constant variable is somewhat of a contradiction - better to use literal reference

**Comment by costin** [comment]
This should now use the newly introduced validation mechanism (see #105477)

**Comment by alex-spies** [comment]
I think this may require [opening up the `PropagateEvalFoldables` optimization rule to aggregations](https://github.com/elastic/elasticsearch/blob/6827f002bb3af3774beecd480b03d1246bd8395d/x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/optimizer/LogicalPlanOptimizer.java#L501) - if so, it might be best if we tackle aggs on consts first (https://github.com/elastic/elasticsearch/issues/100634).

**Comment by fang-xing-esql** [comment]
> I think this may require [opening up the `PropagateEvalFoldables` optimization rule to aggregations](https://github.com/elastic/elasticsearch/blob/6827f002bb3af3774beecd480b03d1246bd8395d/x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/optimizer/LogicalPlanOptimizer.java#L501) - if so, it might be best if we tackle aggs on consts first (#100634).

Indeed, currently PropagateEvalFoldables applies to Filer and Eval plans only, Percentile is under an aggregation plan, blindly apply this rule to aggregate functions doesn't work today. We can either apply this rule to a subset of aggregate functions, certain fields, or support constant in all aggregation functions, whichever is more practical, or a combination of these two.
